src/main/java/org/eclipse/xpanse/terraform/boot/terraform/service/ScriptsGitRepoManage.java

```diff
@@ -60,17 +65,31 @@ public void checkoutScripts(String workspace, TerraformScriptGitRepoDetails scri
         } else {
             log.info("Scripts repo is already cloned in the workspace.");
         }
+        return folderContainsScripts(workspace, scriptsRepo);
     }
 
-
-    /**
-     * Recover method for checkoutScripts.
-     *
-     * @param e GitRepoCloneException
-     */
-    @Recover
-    public void recoverCheckoutScripts(GitRepoCloneException e) {
-        log.error("Retry exhausted. Throwing exception: " + e.getMessage());
-        throw e;
+    private List<File> folderContainsScripts(String workspace,
+                                             TerraformScriptGitRepoDetails scriptsRepo) {
+        File directory = new File(workspace
+                + (StringUtils.isNotBlank(scriptsRepo.getScriptPath())
+                ? File.separator + scriptsRepo.getScriptPath()
+                : ""));
+        File[] files = directory.listFiles();
+        boolean isScriptsExisted = Objects.nonNull(files) && files.length > 0;
+        if (isScriptsExisted) {
+            Optional<File> tfFileOptional = Arrays.stream(files).filter(file ->
+                    file.getName().endsWith(TF_SCRIPT_FILE_EXTENSION)).findAny();
+            isScriptsExisted = tfFileOptional.isPresent();
+        }
+        if (!isScriptsExisted) {
+            throw new GitRepoCloneException(
```

Comment:
Also better we throw another exception. GitRepoCloneException will not make sense here. 
If needed we can create a new custom exception.